src/main/java/de/zalando/aruha/nakadi/controller/PartitionsController.java

```diff
@@ -39,7 +40,11 @@ public PartitionsController(final EventTypeRepository eventTypeRepository, final
         try {
             final EventType eventType = eventTypeRepository.findByName(eventTypeName);
 
-            return ok().body(topicRepository.listPartitions(eventType.getTopic()));
+            if (!topicRepository.topicExists(eventType.getTopic())) {
+                return create(Problem.valueOf(INTERNAL_SERVER_ERROR, "topic is absent in kafka"), request);
```

Comment:
Sometimes I think it's valuable to log 500 errors. It could also be a good idea to monitor then afterwards.

What do you think?